File: app/src/test/java/com/nicholasworkshop/tinklabstest/external/ads/AdsServiceIntegrationTest.java
package com.nicholasworkshop.tinklabstest.external.ads;

import com.nicholasworkshop.tinklabstest.external.ads.model.Ad;

import org.junit.Before;
import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertEquals;

/**
 * Created by nicholas.wong on 2017/05/17.
 */
public class AdsServiceIntegrationTest {

    private AdsService adsService;

    @Before
    public void setUp() throws Exception {
        AdsServiceModule module = new AdsServiceModule();
        adsService = module.adsService();
    }

    @Test
    public void get() throws Exception {
        List<Ad> adList = adsService.get().blockingFirst();
        assertEquals(2, adList.size());
    }
}

File: app/src/test/kotlin/com/nicholasworkshop/tinklabstest/external/ads/AdsServiceIntegrationTest.kt
package com.nicholasworkshop.tinklabstest.external.ads

import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test

class AdsServiceIntegrationTest {

    private lateinit var adsService: AdsService

    @Before
    fun setUp() {
        val module = AdsServiceModule()
        adsService = module.adsService()
    }

    @Test
    fun get() {
        val adList = adsService.get().blockingFirst()
        assertEquals(2, adList.size.toLong())
    }
}